test/test_torch.py

```diff
@@ -7535,6 +7535,18 @@ def test_flip(self, device):
         self.assertEqual(torch.tensor([7, 8, 5, 6, 3, 4, 1, 2]).view(2, 2, 2), data.flip(0, 1))
         self.assertEqual(torch.tensor([8, 7, 6, 5, 4, 3, 2, 1]).view(2, 2, 2), data.flip(0, 1, 2))
 
+        # Complex Dtypes
+        def test_complex_flip(dtype):
```

Comment:
I think we can just update the existing test [here](https://github.com/pytorch/pytorch/blob/08304ccccc11e1a2152d52a8c74b5e88e71c2e6e/test/test_torch.py#L7529)

```
@dtypes(torch.float, torch.cfloat)
def test_flip(self, device, dtype):
        data = torch.randn(8, dtype=dtype, device=device).view(2, 2, 2)
        ...
```

Reply:
The added test is actually a nested function in the mentioned test itself. I'll rename it to `_test_complex_flip` so that it doesn't seem to be an independent test from its name.

Reply:
My bad didn't read the comment carefully. Will do as suggested.

Reply:
Actually using `dtype`, would mean a lot of changes to the test. Every place where `torch.tensor` is initialized to list will have to add `dtype`. Also `arange` doesn't support `complex` dtypes so will have to add `.to(dtype)`
https://github.com/pytorch/pytorch/blob/08304ccccc11e1a2152d52a8c74b5e88e71c2e6e/test/test_torch.py#L7529-L7584

Instead we can have a simple nested function in the test.
```python

        # Complex Dtypes
        def _test_complex_flip(dtype):
            complex_data = torch.tensor([1, 2, 3, 4, 5, 6, 7, 8], device=device, dtype=dtype).view(2, 2, 2)
            self.assertEqual(torch.tensor([5, 6, 7, 8, 1, 2, 3, 4], dtype=dtype).view(2, 2, 2), complex_data.flip(0))
            self.assertEqual(torch.tensor([3, 4, 1, 2, 7, 8, 5, 6], dtype=dtype).view(2, 2, 2), complex_data.flip(1))
            self.assertEqual(torch.tensor([2, 1, 4, 3, 6, 5, 8, 7], dtype=dtype).view(2, 2, 2), complex_data.flip(2))
            self.assertEqual(torch.tensor([7, 8, 5, 6, 3, 4, 1, 2], dtype=dtype).view(2, 2, 2), complex_data.flip(0, 1))
            self.assertEqual(torch.tensor([8, 7, 6, 5, 4, 3, 2, 1], dtype=dtype).view(2, 2, 2), complex_data.flip(0, 1, 2))

        _test_complex_flip(torch.complex64)
        _test_complex_flip(torch.complex128)
```

Do let me know what you feel is a better approach.

Reply:
hmm I guess then we can just add separate tests in `TestTorchDeviceType`:
```
@dtypes(torch.cfloat, torch.cdouble)
def test_complex_flip(dtype):
     complex_data = torch.randn(8, dtype=dtype, device=device).view(2, 2, 2)
     ...
```
and similar thing for rot90